nlp_primitives/number_of_hashtags.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double, NaturalLanguage
+
+from .count_string import CountString
+
+
+class NumberOfHashtags(CountString):
+    """Determines the number of hashtags in a string.
+
+    Description:
+        Given list of strings, determine the number of hashtags
+        in each string. A hashtag is defined as any word starting
+        with a '#' sign followed by a sequence of alphanumeric characters
+        or underscores.
+
+        If a string is missing, return `NaN`.
+
+    Examples:
+        >>> x = ['#regular#expression', 'this is a string', '###__regular#1and_0#expression']
```

Comment:
I guess we can probably debate what a hash tag is exactly, but I believe the `#` symbol that starts a hash tag should be proceeded by a space (or be at the first character position of a the string), and I believe only alpha-numeric characters should be present after the `#` until the space or last character position of the string is reached.

Based on that, I don't think I would call any of these examples valid hashtags.

Did some brief testing and I don't think Twitter recognizes two grouped tags with no space as valid hashtags (`#regular#expression`). It does seem to allow `_` characters though.

My suggestion for the regex would be this:
- `#` symbol must be at first position of the string or preceded by a space
- `#` symbol must be followed by one or more alpha-numeric characters
- Should be terminated by punctuation, a space, or end of string